code/go/0chain.net/blobbercore/writemarker/worker.go

```diff
@@ -16,52 +16,80 @@ func SetupWorkers(ctx context.Context) {
 	go startRedeemWriteMarkers(ctx)
 }
 
-func RedeemMarkersForAllocation(ctx context.Context, allocationObj *allocation.Allocation) error {
-	rctx := datastore.GetStore().CreateTransaction(ctx)
-	db := datastore.GetStore().GetTransaction(rctx)
+func redeemWriterMarkersForAllocation(allocationObj *allocation.Allocation) {
 	defer func() {
-		err := db.Commit().Error
-		if err != nil {
-			Logger.Error("Error committing the writemarker redeem", zap.Error(err))
+		if r := recover(); r != nil {
+			logging.Logger.Error("[recover] redeemWriterMarkersForAllocation", zap.Any("err", r))
 		}
-		rctx.Done()
 	}()
 
-	writemarkers := make([]*WriteMarkerEntity, 0)
+	ctx := datastore.GetStore().CreateTransaction(context.TODO())
+	db := datastore.GetStore().GetTransaction(ctx)
+	var err error
 
-	err := db.Not(WriteMarkerEntity{Status: Committed}).
+	done := false
```

Comment:
Its better to use variable name `shouldRollBack` rather than `done`

Reply:
It was, but it is changed with `done` finally. you have to set it with true on all return if it is named with `shouldRollback`. I would suggest to leave as it is now.